core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -91,9 +89,9 @@ class TransactionStateManager(brokerId: Int,
   // we will get the lock when actually trying to transit the transaction metadata to abort later.
   def transactionsToExpire(): Iterable[TransactionalIdAndProducerIdEpoch] = {
     val now = time.milliseconds()
-    transactionMetadataCache.flatMap { case (_, entry) =>
+    transactionMetadataCache.flatMap { case (txnPartition, entry) =>
```

Comment:
As commented above, this is just to generate a "candidate list" of txns that can be expired, they may not contain all the validate txns nor does every txn included here is expirable. That is why we are not returning the object itself but just the `pid, txnid, epoch` fields and then tries to get the object iteratively again in the `transaction coordinator`.